lms/static/scripts/frontend_apps/components/dashboard/StudentsActivity.tsx

```diff
@@ -6,45 +6,44 @@ import {
 } from '@hypothesis/frontend-shared';
 import type { DataTableProps } from '@hypothesis/frontend-shared';
 import { useState } from 'preact/hooks';
+import { useParams } from 'wouter-preact';
 
-import type { StudentStats } from '../../api-types';
+import type { Assignment, StudentStats } from '../../api-types';
+import { useConfig } from '../../config';
+import { useAPIFetch } from '../../utils/api';
 import { formatDateTime } from '../../utils/date';
 
-export type AssignmentInfo = {
-  title: string;
-};
-
-export type StudentsActivityTableProps = {
-  assignment: AssignmentInfo;
-  students: StudentStats[];
-  loading?: boolean;
-};
-
 type MandatoryOrder<T> = NonNullable<DataTableProps<T>['order']>;
 
-export default function StudentsActivityTable({
-  assignment,
-  students,
-  loading,
-}: StudentsActivityTableProps) {
-  const title = `Assignment: ${assignment.title}`;
+export default function StudentsActivity() {
+  const { dashboard } = useConfig(['dashboard']);
+  const { routes } = dashboard;
+  const { assignmentId } = useParams<{ assignmentId: string }>();
+  const assignment = useAPIFetch<Assignment>(
+    routes.assignment.replace(':assignment_id', assignmentId),
+  );
+  const students = useAPIFetch<StudentStats[]>(
+    routes.assignment_stats.replace(':assignment_id', assignmentId),
+  );
+
+  const title = `Assignment: ${assignment.data?.title}`;
   const [order, setOrder] = useState<MandatoryOrder<StudentStats>>({
     field: 'display_name',
     direction: 'ascending',
   });
-  const orderedStudents = useOrderedRows(students, order);
+  const orderedStudents = useOrderedRows(students.data ?? [], order);
 
   return (
     <Card>
       <CardContent>
         <h2 className="text-brand mb-3 text-xl" data-testid="title">
-          {title}
+          {assignment.isLoading ? 'Loading...' : title}
```

Comment:
The text transition creates a flicker when the title becomes available. You want to render something more subtle that takes up the same amount of vertical space.

Also, you need to handle the error state. If the title fails to load, you get:

<img width="822" alt="assignment-undefined" src="https://github.com/hypothesis/lms/assets/2458/c2039ad5-6bce-4428-bcfe-a0afd41a015e">

The API of `useAPIFetch` was designed modeled after [SWR](https://swr.vercel.app), but one downside of their approach is that you are not forced to handle loading / error states, as you would be if the result was something like `{ state: 'loading' } | { state: 'ok', data: T } | { state: 'error', data: Error }`

Reply:
Good catch.

Out of curiosity, did you really get an error when fetching that info, or is it something you faked in order to test non-happy paths?

Reply:
I faked an error on the backend.

Reply:
The backed never fails, typescript might point that out.

Related to this I still have to fix the BE returning HTML content for 403 responses in 
API endpoints.

Reply:
> The backed never fails, typescript might point that out.

We can model the state as `{ state: 'loading' } | { state: 'ok', data: T }` and leave `{ state: 'error', data: Error }` out as a ridiculously improbable edge case 😂